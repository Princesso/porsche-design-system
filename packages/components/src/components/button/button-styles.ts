import type { BreakpointCustomizable, ButtonVariant, Theme, LinkButtonIconName } from '../../types';
import { getCss } from '../../utils';
import { getLinkButtonStyles } from '../../styles/link-button-styles';

export const getComponentCss = (
  icon: LinkButtonIconName,
  iconSource: string,
  variant: ButtonVariant,
  hideLabel: BreakpointCustomizable<boolean>,
  isDisabledOrLoading: boolean,
  theme: Theme
): string => {
  return getCss(getLinkButtonStyles(icon, iconSource, variant, hideLabel, isDisabledOrLoading, false, theme));
};
