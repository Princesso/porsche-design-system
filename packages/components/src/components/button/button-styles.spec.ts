import { getComponentCss } from './button-styles';

xdescribe('getComponentCss()', () => {
  it.each<Parameters<typeof getComponentCss>>([
    ['arrow-right', '', 'primary', false, false, false, false, 'light'],
    ['arrow-right', '', 'primary', false, false, false, false, 'dark'],
    ['arrow-right', '', 'secondary', false, false, false, false, 'light'],
    ['arrow-right', '', 'secondary', false, false, false, false, 'dark'],
    ['arrow-right', '', 'tertiary', false, false, false, false, 'light'],
    ['arrow-right', '', 'tertiary', false, false, false, false, 'dark'],
    ['arrow-right', '', 'primary', false, true, false, false, 'light'],
    ['arrow-right', '', 'primary', false, false, true, false, 'light'],
    ['arrow-right', '', 'primary', false, false, false, true, 'light'],
    ['arrow-right', '', 'primary', { base: true, xs: false, s: true, m: false, l: true, xl: false }, true,false, false, 'dark'],
  ])('should return correct css for icon: %s, iconSource: %s, variant: %s, hideLabel: %s, disabled: %s, isDisabledOrLoading: %s, loading: %s and theme: %s', (...args) => {
    expect(getComponentCss(...args)).toMatchSnapshot();
  });
});
