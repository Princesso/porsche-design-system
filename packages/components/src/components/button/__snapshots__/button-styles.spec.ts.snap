// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getComponentCss() should return correct css for icon: arrow-right, iconSource: , variant: primary, hideLabel: { base: true, xs: false, s: true, m: false, l: true, xl: false }, disabled: true, loading: false and theme: dark 1`] = `
":host {
  display: inline-block;
  vertical-align: top;
  transform: translate3d(0,0,0);
  outline: 0 !important;
}
.root {
  display: flex;
  align-items: flex-start;
  justify-content: center;
  width: 100%;
  min-width: 54px;
  min-height: 54px;
  box-sizing: border-box;
  outline: 0;
  text-align: left;
  appearance: none;
  text-decoration: none;
  border: 2px solid #FBFCFF;
  border-radius: 4px;
  background-color: #7E7F82;
  color: #AFB0B3;
  font: normal normal 400 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
  transition: background-color var(--p-transition-duration, .24s) ease,border-color var(--p-transition-duration, .24s) ease,color var(--p-transition-duration, .24s) ease;
  padding: 13px;
  gap: 0;
  cursor: not-allowed;
  border-color: #7E7F82;
}
.root:focus::before {
  content: "";
  position: fixed;
  border: 2px solid #0A0AFF;
  border-radius: 8px;
  top: -4px;
  right: -4px;
  bottom: -4px;
  left: -4px;
}
.root:not(:focus-visible)::before {
  border: 0;
}
.label {
  width: 0;
  height: 1px;
  text-indent: -999999px;
  transition: opacity var(--p-transition-duration, .24s) ease;
}
.icon {
  width: calc(6px + 2.125ex);
  height: calc(6px + 2.125ex);
  pointer-events: none;
  margin-left: 0;
  transition: opacity var(--p-transition-duration, .24s) ease;
}
@media(min-width:480px) {
  .root {
    padding: 13px 26px;
    gap: 8px;
  }
  .label {
    width: auto;
    height: auto;
    text-indent: 0;
  }
  .icon {
    margin-left: -8px;
  }
}

@media(min-width:760px) {
  .root {
    padding: 13px;
    gap: 0;
  }
  .label {
    width: 0;
    height: 1px;
    text-indent: -999999px;
  }
  .icon {
    margin-left: 0;
  }
}

@media(min-width:1000px) {
  .root {
    padding: 13px 26px;
    gap: 8px;
  }
  .label {
    width: auto;
    height: auto;
    text-indent: 0;
  }
  .icon {
    margin-left: -8px;
  }
}

@media(min-width:1300px) {
  .root {
    padding: 13px;
    gap: 0;
  }
  .label {
    width: 0;
    height: 1px;
    text-indent: -999999px;
  }
  .icon {
    margin-left: 0;
  }
}

@media(min-width:1760px) {
  .root {
    padding: 13px 26px;
    gap: 8px;
  }
  .label {
    width: auto;
    height: auto;
    text-indent: 0;
  }
  .icon {
    margin-left: -8px;
  }
}
"
`;

exports[`getComponentCss() should return correct css for icon: arrow-right, iconSource: , variant: primary, hideLabel: false, disabled: false, loading: false and theme: dark 1`] = `
":host {
  display: inline-block;
  vertical-align: top;
  transform: translate3d(0,0,0);
  outline: 0 !important;
}
.root {
  display: flex;
  align-items: flex-start;
  justify-content: center;
  width: 100%;
  min-width: 54px;
  min-height: 54px;
  box-sizing: border-box;
  outline: 0;
  text-align: left;
  appearance: none;
  text-decoration: none;
  border: 2px solid #FBFCFF;
  border-radius: 4px;
  background-color: #FBFCFF;
  color: #010205;
  font: normal normal 400 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
  transition: background-color var(--p-transition-duration, .24s) ease,border-color var(--p-transition-duration, .24s) ease,color var(--p-transition-duration, .24s) ease;
  padding: 13px 26px;
  gap: 8px;
  cursor: pointer;
}
.root:focus::before {
  content: "";
  position: fixed;
  border: 2px solid #0A0AFF;
  border-radius: 8px;
  top: -4px;
  right: -4px;
  bottom: -4px;
  left: -4px;
}
.root:not(:focus-visible)::before {
  border: 0;
}
.label {
  width: auto;
  height: auto;
  text-indent: 0;
  transition: opacity var(--p-transition-duration, .24s) ease;
}
.icon {
  width: calc(6px + 2.125ex);
  height: calc(6px + 2.125ex);
  pointer-events: none;
  margin-left: -8px;
  transition: opacity var(--p-transition-duration, .24s) ease;
}
@media(hover:hover) {
  .root:hover {
    background-color: #AFB0B3;
    border-color: #AFB0B3;
  }
}
"
`;

exports[`getComponentCss() should return correct css for icon: arrow-right, iconSource: , variant: primary, hideLabel: false, disabled: false, loading: false and theme: light 1`] = `
":host {
  display: inline-block;
  vertical-align: top;
  transform: translate3d(0,0,0);
  outline: 0 !important;
}
.root {
  display: flex;
  align-items: flex-start;
  justify-content: center;
  width: 100%;
  min-width: 54px;
  min-height: 54px;
  box-sizing: border-box;
  outline: 0;
  text-align: left;
  appearance: none;
  text-decoration: none;
  border: 2px solid #010205;
  border-radius: 4px;
  background-color: #010205;
  color: #FBFCFF;
  font: normal normal 400 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
  transition: background-color var(--p-transition-duration, .24s) ease,border-color var(--p-transition-duration, .24s) ease,color var(--p-transition-duration, .24s) ease;
  padding: 13px 26px;
  gap: 8px;
  cursor: pointer;
}
.root:focus::before {
  content: "";
  position: fixed;
  border: 2px solid #0A0AFF;
  border-radius: 8px;
  top: -4px;
  right: -4px;
  bottom: -4px;
  left: -4px;
}
.root:not(:focus-visible)::before {
  border: 0;
}
.label {
  width: auto;
  height: auto;
  text-indent: 0;
  transition: opacity var(--p-transition-duration, .24s) ease;
}
.icon {
  width: calc(6px + 2.125ex);
  height: calc(6px + 2.125ex);
  pointer-events: none;
  margin-left: -8px;
  transition: opacity var(--p-transition-duration, .24s) ease;
}
@media(hover:hover) {
  .root:hover {
    background-color: #535457;
    border-color: #535457;
  }
}
"
`;

exports[`getComponentCss() should return correct css for icon: arrow-right, iconSource: , variant: primary, hideLabel: false, disabled: false, loading: false and theme: light 2`] = `
":host {
  display: inline-block;
  vertical-align: top;
  transform: translate3d(0,0,0);
  outline: 0 !important;
}
.root {
  display: flex;
  align-items: flex-start;
  justify-content: center;
  width: 100%;
  min-width: 54px;
  min-height: 54px;
  box-sizing: border-box;
  outline: 0;
  text-align: left;
  appearance: none;
  text-decoration: none;
  border: 2px solid #010205;
  border-radius: 4px;
  background-color: #010205;
  color: #FBFCFF;
  font: normal normal 400 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
  transition: background-color var(--p-transition-duration, .24s) ease,border-color var(--p-transition-duration, .24s) ease,color var(--p-transition-duration, .24s) ease;
  padding: 13px 26px;
  gap: 8px;
  cursor: pointer;
}
.root:focus::before {
  content: "";
  position: fixed;
  border: 2px solid #0A0AFF;
  border-radius: 8px;
  top: -4px;
  right: -4px;
  bottom: -4px;
  left: -4px;
}
.root:not(:focus-visible)::before {
  border: 0;
}
.label {
  width: auto;
  height: auto;
  text-indent: 0;
  transition: opacity var(--p-transition-duration, .24s) ease;
}
.icon {
  width: calc(6px + 2.125ex);
  height: calc(6px + 2.125ex);
  pointer-events: none;
  margin-left: -8px;
  transition: opacity var(--p-transition-duration, .24s) ease;
}
@media(hover:hover) {
  .root:hover {
    background-color: #535457;
    border-color: #535457;
  }
}
"
`;

exports[`getComponentCss() should return correct css for icon: arrow-right, iconSource: , variant: primary, hideLabel: false, disabled: false, loading: true and theme: light 1`] = `
":host {
  display: inline-block;
  vertical-align: top;
  transform: translate3d(0,0,0);
  outline: 0 !important;
}
.root {
  display: flex;
  align-items: flex-start;
  justify-content: center;
  width: 100%;
  min-width: 54px;
  min-height: 54px;
  box-sizing: border-box;
  outline: 0;
  text-align: left;
  appearance: none;
  text-decoration: none;
  border: 2px solid #010205;
  border-radius: 4px;
  background-color: #535457;
  color: #535457;
  font: normal normal 400 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
  transition: background-color var(--p-transition-duration, .24s) ease,border-color var(--p-transition-duration, .24s) ease,color var(--p-transition-duration, .24s) ease;
  padding: 13px 26px;
  gap: 8px;
  cursor: not-allowed;
  border-color: #535457;
}
.root:focus::before {
  content: "";
  position: fixed;
  border: 2px solid #0A0AFF;
  border-radius: 8px;
  top: -4px;
  right: -4px;
  bottom: -4px;
  left: -4px;
}
.root:not(:focus-visible)::before {
  border: 0;
}
.label {
  width: auto;
  height: auto;
  text-indent: 0;
  transition: opacity var(--p-transition-duration, .24s) ease;
  opacity: 0;
}
.icon {
  width: calc(6px + 2.125ex);
  height: calc(6px + 2.125ex);
  pointer-events: none;
  margin-left: -8px;
  transition: opacity var(--p-transition-duration, .24s) ease;
  opacity: 0;
}
.spinner {
  width: calc(6px + 2.125ex);
  height: calc(6px + 2.125ex);
  pointer-events: none;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}"
`;

exports[`getComponentCss() should return correct css for icon: arrow-right, iconSource: , variant: primary, hideLabel: false, disabled: true, loading: false and theme: light 1`] = `
":host {
  display: inline-block;
  vertical-align: top;
  transform: translate3d(0,0,0);
  outline: 0 !important;
}
.root {
  display: flex;
  align-items: flex-start;
  justify-content: center;
  width: 100%;
  min-width: 54px;
  min-height: 54px;
  box-sizing: border-box;
  outline: 0;
  text-align: left;
  appearance: none;
  text-decoration: none;
  border: 2px solid #010205;
  border-radius: 4px;
  background-color: #949598;
  color: #535457;
  font: normal normal 400 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
  transition: background-color var(--p-transition-duration, .24s) ease,border-color var(--p-transition-duration, .24s) ease,color var(--p-transition-duration, .24s) ease;
  padding: 13px 26px;
  gap: 8px;
  cursor: not-allowed;
  border-color: #949598;
}
.root:focus::before {
  content: "";
  position: fixed;
  border: 2px solid #0A0AFF;
  border-radius: 8px;
  top: -4px;
  right: -4px;
  bottom: -4px;
  left: -4px;
}
.root:not(:focus-visible)::before {
  border: 0;
}
.label {
  width: auto;
  height: auto;
  text-indent: 0;
  transition: opacity var(--p-transition-duration, .24s) ease;
}
.icon {
  width: calc(6px + 2.125ex);
  height: calc(6px + 2.125ex);
  pointer-events: none;
  margin-left: -8px;
  transition: opacity var(--p-transition-duration, .24s) ease;
}"
`;

exports[`getComponentCss() should return correct css for icon: arrow-right, iconSource: , variant: secondary, hideLabel: false, disabled: false, loading: false and theme: dark 1`] = `
":host {
  display: inline-block;
  vertical-align: top;
  transform: translate3d(0,0,0);
  outline: 0 !important;
}
.root {
  display: flex;
  align-items: flex-start;
  justify-content: center;
  width: 100%;
  min-width: 54px;
  min-height: 54px;
  box-sizing: border-box;
  outline: 0;
  text-align: left;
  appearance: none;
  text-decoration: none;
  border: 2px solid #FBFCFF;
  border-radius: 4px;
  background-color: transparent;
  color: #FBFCFF;
  font: normal normal 400 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
  transition: background-color var(--p-transition-duration, .24s) ease,border-color var(--p-transition-duration, .24s) ease,color var(--p-transition-duration, .24s) ease;
  padding: 13px 26px;
  gap: 8px;
  cursor: pointer;
}
.root:focus::before {
  content: "";
  position: fixed;
  border: 2px solid #0A0AFF;
  border-radius: 8px;
  top: -4px;
  right: -4px;
  bottom: -4px;
  left: -4px;
}
.root:not(:focus-visible)::before {
  border: 0;
}
.label {
  width: auto;
  height: auto;
  text-indent: 0;
  transition: opacity var(--p-transition-duration, .24s) ease;
}
.icon {
  width: calc(6px + 2.125ex);
  height: calc(6px + 2.125ex);
  pointer-events: none;
  margin-left: -8px;
  transition: opacity var(--p-transition-duration, .24s) ease;
}
@media(hover:hover) {
  .root:hover {
    background-color: rgba(126, 127, 130, 0.20);
    border-color: #7E7F82;
    -webkit-backdrop-filter: blur(32px);
    backdrop-filter: blur(32px);
  }
  .root:hover:focus::before {
    top: -6px;
    right: -6px;
    bottom: -6px;
    left: -6px;
  }
  .root:hover > .label {
    color: #FBFCFF;
  }
  .root:hover > .icon {
    color: #FBFCFF;
  }
}
"
`;

exports[`getComponentCss() should return correct css for icon: arrow-right, iconSource: , variant: secondary, hideLabel: false, disabled: false, loading: false and theme: light 1`] = `
":host {
  display: inline-block;
  vertical-align: top;
  transform: translate3d(0,0,0);
  outline: 0 !important;
}
.root {
  display: flex;
  align-items: flex-start;
  justify-content: center;
  width: 100%;
  min-width: 54px;
  min-height: 54px;
  box-sizing: border-box;
  outline: 0;
  text-align: left;
  appearance: none;
  text-decoration: none;
  border: 2px solid #010205;
  border-radius: 4px;
  background-color: transparent;
  color: #010205;
  font: normal normal 400 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
  transition: background-color var(--p-transition-duration, .24s) ease,border-color var(--p-transition-duration, .24s) ease,color var(--p-transition-duration, .24s) ease;
  padding: 13px 26px;
  gap: 8px;
  cursor: pointer;
}
.root:focus::before {
  content: "";
  position: fixed;
  border: 2px solid #0A0AFF;
  border-radius: 8px;
  top: -4px;
  right: -4px;
  bottom: -4px;
  left: -4px;
}
.root:not(:focus-visible)::before {
  border: 0;
}
.label {
  width: auto;
  height: auto;
  text-indent: 0;
  transition: opacity var(--p-transition-duration, .24s) ease;
}
.icon {
  width: calc(6px + 2.125ex);
  height: calc(6px + 2.125ex);
  pointer-events: none;
  margin-left: -8px;
  transition: opacity var(--p-transition-duration, .24s) ease;
}
@media(hover:hover) {
  .root:hover {
    background-color: rgba(148, 149, 152, 0.20);
    border-color: #949598;
    -webkit-backdrop-filter: blur(32px);
    backdrop-filter: blur(32px);
  }
  .root:hover:focus::before {
    top: -6px;
    right: -6px;
    bottom: -6px;
    left: -6px;
  }
  .root:hover > .label {
    color: #010205;
  }
  .root:hover > .icon {
    color: #010205;
  }
}
"
`;

exports[`getComponentCss() should return correct css for icon: arrow-right, iconSource: , variant: tertiary, hideLabel: false, disabled: false, loading: false and theme: dark 1`] = `
":host {
  display: inline-block;
  vertical-align: top;
  transform: translate3d(0,0,0);
  outline: 0 !important;
}
.root {
  display: flex;
  align-items: flex-start;
  justify-content: center;
  width: 100%;
  min-width: 54px;
  min-height: 54px;
  box-sizing: border-box;
  outline: 0;
  text-align: left;
  appearance: none;
  text-decoration: none;
  border: 2px solid #FBFCFF;
  border-radius: 4px;
  background-color: transparent;
  color: #FBFCFF;
  font: normal normal 400 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
  transition: background-color var(--p-transition-duration, .24s) ease,border-color var(--p-transition-duration, .24s) ease,color var(--p-transition-duration, .24s) ease;
  padding: 13px 26px;
  gap: 8px;
  cursor: pointer;
}
.root:focus::before {
  content: "";
  position: fixed;
  border: 2px solid #0A0AFF;
  border-radius: 8px;
  top: -4px;
  right: -4px;
  bottom: -4px;
  left: -4px;
}
.root:not(:focus-visible)::before {
  border: 0;
}
.label {
  width: auto;
  height: auto;
  text-indent: 0;
  transition: opacity var(--p-transition-duration, .24s) ease;
}
.icon {
  width: calc(6px + 2.125ex);
  height: calc(6px + 2.125ex);
  pointer-events: none;
  margin-left: -8px;
  transition: opacity var(--p-transition-duration, .24s) ease;
}
@media(hover:hover) {
  .root:hover {
    background-color: rgba(126, 127, 130, 0.20);
    border-color: #7E7F82;
    -webkit-backdrop-filter: blur(32px);
    backdrop-filter: blur(32px);
  }
  .root:hover:focus::before {
    top: -6px;
    right: -6px;
    bottom: -6px;
    left: -6px;
  }
  .root:hover > .label {
    color: #FBFCFF;
  }
  .root:hover > .icon {
    color: #FBFCFF;
  }
}
"
`;

exports[`getComponentCss() should return correct css for icon: arrow-right, iconSource: , variant: tertiary, hideLabel: false, disabled: false, loading: false and theme: light 1`] = `
":host {
  display: inline-block;
  vertical-align: top;
  transform: translate3d(0,0,0);
  outline: 0 !important;
}
.root {
  display: flex;
  align-items: flex-start;
  justify-content: center;
  width: 100%;
  min-width: 54px;
  min-height: 54px;
  box-sizing: border-box;
  outline: 0;
  text-align: left;
  appearance: none;
  text-decoration: none;
  border: 2px solid #010205;
  border-radius: 4px;
  background-color: transparent;
  color: #010205;
  font: normal normal 400 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
  transition: background-color var(--p-transition-duration, .24s) ease,border-color var(--p-transition-duration, .24s) ease,color var(--p-transition-duration, .24s) ease;
  padding: 13px 26px;
  gap: 8px;
  cursor: pointer;
}
.root:focus::before {
  content: "";
  position: fixed;
  border: 2px solid #0A0AFF;
  border-radius: 8px;
  top: -4px;
  right: -4px;
  bottom: -4px;
  left: -4px;
}
.root:not(:focus-visible)::before {
  border: 0;
}
.label {
  width: auto;
  height: auto;
  text-indent: 0;
  transition: opacity var(--p-transition-duration, .24s) ease;
}
.icon {
  width: calc(6px + 2.125ex);
  height: calc(6px + 2.125ex);
  pointer-events: none;
  margin-left: -8px;
  transition: opacity var(--p-transition-duration, .24s) ease;
}
@media(hover:hover) {
  .root:hover {
    background-color: rgba(148, 149, 152, 0.20);
    border-color: #949598;
    -webkit-backdrop-filter: blur(32px);
    backdrop-filter: blur(32px);
  }
  .root:hover:focus::before {
    top: -6px;
    right: -6px;
    bottom: -6px;
    left: -6px;
  }
  .root:hover > .label {
    color: #010205;
  }
  .root:hover > .icon {
    color: #010205;
  }
}
"
`;

exports[`getComponentCss() should return correct css for icon: none, iconSource: , variant: primary, hideLabel: false, disabled: true, loading: true and theme: light 1`] = `
":host {
  display: inline-block;
  vertical-align: top;
  transform: translate3d(0,0,0);
  outline: 0 !important;
}
.root {
  display: flex;
  align-items: flex-start;
  justify-content: center;
  width: 100%;
  min-width: 54px;
  min-height: 54px;
  box-sizing: border-box;
  outline: 0;
  text-align: left;
  appearance: none;
  text-decoration: none;
  border: 2px solid #010205;
  border-radius: 4px;
  background-color: #535457;
  color: #535457;
  font: normal normal 400 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
  transition: background-color var(--p-transition-duration, .24s) ease,border-color var(--p-transition-duration, .24s) ease,color var(--p-transition-duration, .24s) ease;
  padding: 13px 26px;
  gap: 8px;
  cursor: not-allowed;
  border-color: #535457;
}
.root:focus::before {
  content: "";
  position: fixed;
  border: 2px solid #0A0AFF;
  border-radius: 8px;
  top: -4px;
  right: -4px;
  bottom: -4px;
  left: -4px;
}
.root:not(:focus-visible)::before {
  border: 0;
}
.label {
  width: auto;
  height: auto;
  text-indent: 0;
  transition: opacity var(--p-transition-duration, .24s) ease;
  opacity: 0;
}
.spinner {
  width: calc(6px + 2.125ex);
  height: calc(6px + 2.125ex);
  pointer-events: none;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}
.icon {
  transition: opacity var(--p-transition-duration, .24s) ease;
  opacity: 0;
}"
`;
